stats/view/worker.go

```diff
@@ -187,41 +188,52 @@ func (w *worker) stop() {
 	<-w.done
 }
 
-func (w *worker) getMeasureRef(m stats.Measure) *measureRef {
-	if mr, ok := w.measures[m.Name()]; ok {
+func (w *worker) getMeasureRef(name string) *measureRef {
+	if mr, ok := w.measures[name]; ok {
 		return mr
 	}
 	mr := &measureRef{
-		measure: m,
-		views:   make(map[*View]struct{}),
+		measure: name,
+		views:   make(map[*viewInternal]struct{}),
 	}
-	w.measures[m.Name()] = mr
+	w.measures[name] = mr
 	return mr
 }
 
-func (w *worker) tryRegisterView(v *View) error {
-	if err := checkViewName(v.name); err != nil {
-		return err
+func (w *worker) tryRegisterView(v *View) (*viewInternal, error) {
+	if err := checkViewName(v.Name); err != nil {
+		return nil, err
 	}
-	if x, ok := w.views[v.Name()]; ok {
-		if x != v {
-			return fmt.Errorf("cannot register view %q; another view with the same name is already registered", v.Name())
+	sort.Slice(v.GroupByTags, func(i, j int) bool {
+		return v.GroupByTags[i].Name() < v.GroupByTags[j].Name()
+	})
+	if x, ok := w.views[v.Name]; ok {
+		if !x.definition.Equal(v) {
+			return nil, fmt.Errorf("cannot subscribe view %q; a different view with the same name is already subscribed", v.Name)
 		}
 
 		// the view is already registered so there is nothing to do and the
 		// command is considered successful.
-		return nil
+		return x, nil
 	}
 
-	if v.Measure() == nil {
-		return fmt.Errorf("cannot register view %q: measure not defined", v.Name())
+	if v.MeasureName == "" {
+		return nil, fmt.Errorf("cannot subscribe view %q: measure not set", v.Name)
 	}
+	m := stats.FindMeasure(v.MeasureName)
```

Comment:
My point about v.MeasureName applies here given that a user invoke m.Name() to create *view.View.MeasureName and now here we are looking for Measure again